### PR TITLE
Add support for pip-compile's `--unsafe-package` flag

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

In uv, we're going to use `--no-emit-package` for this, to convey that the package will be included in the resolution but not in the output file. It also mirrors flags like `--emit-index-url`.

We're also including an `--unsafe-package` alias.

Closes https://github.com/astral-sh/uv/issues/1415.

### CODE REVIEW

**Comment by charliermarsh** [comment]
I'm torn on the name `--unsafe-package`, and the use of "unsafe" in the footer comment (which matches pip-compile), like:

```
# The following packages are considered to be unsafe in a requirements file:
# jinja2
```

**Comment by hauntsaninja** [comment]
Yeah :-/ It's unfortunate that pip-compile puts those words in the file. I do think it'd be nice to have an `--exclude-package` alias.

File: crates/uv/src/commands/pip_compile.rs [hunk 1]
```diff
@@ -383,12 +384,30 @@ pub(crate) async fn pip_compile(
...
+    let excluded = no_emit_packages
+        .into_iter()
+        .filter(|name| resolution.contains(name))
+        .collect::<Vec<_>>();
+    if !excluded.is_empty() {
+        writeln!(writer)?;
+        writeln!(
+            writer,
+            "{}",
+            "# The following packages were included while generating the resolution:".green()
```
Comment (hauntsaninja) [inline crates/uv/src/commands/pip_compile.rs:404]: Thank you for implementing the feature!

nit: I was confused by "include" here, maybe `The following packages were omitted from the result:` (although maybe omit vs no-emit becomes confusing too, so maybe just excluded)

Comment (charliermarsh) [reply crates/uv/src/commands/pip_compile.rs:404]: Good idea -- changed in https://github.com/astral-sh/uv/pull/1935.

**Pull request merged**